File: ContestOfSwift/ContestOfSwift/Challenges/Challenge4.swift
//
//  Challenge4.swift
//  ContestOfSwift
//
//  Created by Sergio Daniel on 4/24/24.
//

/*
 * Escribe un programa que se encargue de comprobar si un número es o no primo.
 * Hecho esto, imprime los números primos entre 1 y 100.
 */

import Foundation

enum Challenge4 {

    static func isPrime(_ number: Int) -> Bool { number.isPrime }

    static func perform() {
        var primeNumbers = [Int]()
        for n in (0...100) {
            if isPrime(n) {
                primeNumbers.append(n)
            }
        }
        print(
            primeNumbers
                .map { "\($0)" }
                .joined(separator: ", ")
        )
    }

}

File: contestOfKotlin/src/main/kotlin/challenges/Challenge4.kt
package org.example.challenges.challenge4

/*
 * Escribe un programa que se encargue de comprobar si un número es o no primo.
 * Hecho esto, imprime los números primos entre 1 y 100.
 */

fun isPrime(number: Int): Boolean {
    if (number < 2) return false

    for (n in 2 until number) {
        if (number % n == 0) return false
    }

    return true
}

fun performPrimeNumbers() {
    val primeNumbers: MutableList<Int> = mutableListOf()
    for (n in 0..100) {
        if (isPrime(n)) {
            primeNumbers.add(n)
        }
    }
    println(
        primeNumbers
            .joinToString(separator = ", ")
    )
}